sdk/storage/Azure.Storage.Blobs/AzureStorageNetMigrationV12.md

```diff
@@ -0,0 +1,268 @@
+
+// Create a client that can authenticate with a connection string
+BlobServiceClient service = new BlobServiceClient(connectionString);
+
+// Make a service request to verify we've successfully authenticated
+await service.GetPropertiesAsync();
+```
+
+### Shared Access Policies
+
+TBD
+
+### Client hierarchy
+
+In the interest of simplifying the API surface we've made a three top level clients that can be used to interact with a majority of your resources: `BlobServiceClient`, `BlobContainerClient`, and `BlobClient`.
+
+[//]: # (Blob Metadata, properties, and attributes...)
+
+### Client constructors
+
+| In v11 | In v12 |
+|-------|--------|
+| `CloudStorageAccount` | `BlobServiceClient` |
+| `CloudBlobContainer`  | `BlobContainerClient` |
+| `CloudBlobClient` | `BlobClient` |
+| `CloudBlockBlob` | `BlockBlobClient` |
+
+### Creating a Container
+
+v11
+```c
+// Create the CloudBlobClient that represents the Blob storage endpoint for the storage account.
+CloudBlobClient cloudBlobClient = storageAccount.CreateCloudBlobClient();
+
+CloudBlobContainer cloudBlobContainer =
+    cloudBlobClient.GetContainerReference("yourcontainer");
+await cloudBlobContainer.CreateAsync();
+```
+
+v12
+```c
+// Create a BlobServiceClient object which will be used to create a container client
+BlobServiceClient blobServiceClient = new BlobServiceClient(connectionString);
+
+// Create a unique name for the container
+string containerName = "yourcontainer";
+
+// Create the container and return a container client object
+BlobContainerClient containerClient = await blobServiceClient.CreateBlobContainerAsync(containerName);
+```
+
+Summary: In version 11, you would have to use the storage account endpoint to create the `CloudBlobClient`. Then you can get the container reference by calling the `GetContainerReference` method and create it by calling `CreateAsync()` on it. In version 12, you will use the `BlobServiceClient` object to then create `BlobContainerClient`.
+
+
+### Uploading Blobs to a Container
+
+v11
+```c
+// Assumes cloudBlobContainer already contains a reference to the container.
+// filename is the intended blob name as a string
+// localFilePath should be the path to the local file you want to upload
+
+// Get a reference to the blob address, then upload the file to the blob.
+CloudBlockBlob cloudBlockBlob = cloudBlobContainer.GetBlockBlobReference(filename);
+await cloudBlockBlob.UploadFromFileAsync(localFilePath);
+```
+
+v12
+```c
+// Assumes containerClient already contains a reference to the container.
+// filename is the intended blob name as a string
+// localFilePath should be the path to the local file you want to upload
+
+// Get a reference to a blob
+BlobClient blobClient = containerClient.GetBlobClient(filename);
+
+// Open the file and upload its data
+using FileStream uploadFileStream = File.OpenRead(localFilePath);
+await blobClient.UploadAsync(uploadFileStream, true);
```

Comment:
What do you think of using a named parameter here to clarify the purpose of the `true`?

```suggestion
await blobClient.UploadAsync(uploadFileStream, overwrite: true);
```

Reply:
done!